File: demos/ios/Demos/DemoViewController.m

#import "DemoViewController.h"

@interface DemoViewController ()
{
    NSString *_type;
    NSString *_name;
    NSString *_viewName;
}
@end

@implementation DemoViewController


-(instancetype)initWithInfo:(NSDictionary *)info {
    if (self = [super init]) {
        _type = [info objectForKey:@"type"];
        _name = [info objectForKey:@"name"];
        _viewName = [info objectForKey:@"view"];
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.edgesForExtendedLayout = UIRectEdgeNone;
    self.navigationController.navigationBar.backgroundColor = [UIColor whiteColor];
    self.view.backgroundColor = [UIColor whiteColor];
    if (@available(iOS 13, *)) {
        UINavigationBarAppearance *appearance = [[UINavigationBarAppearance alloc] init];
        [appearance configureWithOpaqueBackground];
        self.navigationController.navigationBar.standardAppearance = appearance;
        self.navigationController.navigationBar.scrollEdgeAppearance = appearance;
    }
    self.title = _name;
    [self.view addSubview:[self demoView]];
}


- (UIView *)demoView {
    Class class = NSClassFromString(_viewName);
    UIView *demoView = [class alloc];
#if (TARGET_OS_MACCATALYST == 1)
    demoView = [demoView initWithFrame:CGRectMake(0, 0, UIScreen.mainScreen.bounds.size.width, UIScreen.mainScreen.bounds.size.height)];
#else
    demoView = [demoView initWithFrame:CGRectMake(0, 0, UIScreen.mainScreen.bounds.size.width, 280)];
#endif
    return demoView;
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/DemoViewController.swift
//
//  DemoViewController.swift
//  Demos-Swift
//
//  Created by weiqing.twq on 2022/6/1.
//

import UIKit

class DemoViewController: UIViewController {
    
    var info: NSDictionary;
    
    required init?(coder: NSCoder) {
        self.info = NSDictionary.init();
        super.init(coder: coder)
    }
    
    required init(info: NSDictionary) {
        self.info = info;
        super.init(nibName: nil, bundle: nil);
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.edgesForExtendedLayout = UIRectEdge.init(rawValue:0);
        if #available(iOS 13.0, *) {
            let appearance = UINavigationBarAppearance.init()
            appearance.configureWithOpaqueBackground()
            self.navigationController?.navigationBar.standardAppearance = appearance
            self.navigationController?.navigationBar.scrollEdgeAppearance = appearance
        }
     
        self.title = self.info.object(forKey: "name") as? String
        self.navigationController?.navigationBar.backgroundColor = UIColor.white
        self.view.backgroundColor = UIColor.white;
        self.view.addSubview(self.demoView())
    }
    
    func demoView() -> UIView {
        var appName = Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as! String
        appName = appName.replacingOccurrences(of: "-", with: "_")
        let viewName = self.info.object(forKey: "view") as! String;
        let className = appName + "." + viewName;
        let cls = NSClassFromString(className) as! UIView.Type
        let view = cls.init(frame:CGRect(x: 0, y: 0, width: self.view.bounds.width, height: 280))
        return view
    }
}
